### PR TITLE
windows support

### LINKED ISSUE #90: Support Windows
Framework does not currently support Windows (though some users have had success with the [Windows Subsystem for Linux (WSL)](https://learn.microsoft.com/en-us/windows/wsl/install)). It should work, and we are working on it.

### PR DESCRIPTION
**Pull request by mythmon**
Fixes #90.

- **disable autocrlf**
- **test on windows**
- **cross-spawn**
- **rimraf**
- **use cross-env to run mocha**
- **fix tests with posix paths**
- **one less op**
- **mostly posix**
- **fix data loader and config tests**
- **more tests**
- **all tests passing**

### CODE REVIEW

**Review by mbostock — APPROVED** [review]

**Comment by mythmon** [comment]
I'm enabling auto merge on this; but it won't merge until our CI configuration change to look for the newly added CI jobs instead of the old one.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -11,7 +11,9 @@ jobs:
     strategy:
       matrix:
         version: [20, 21]
-    runs-on: ubuntu-latest
+        os: [ubuntu-latest, windows-latest]
+      fail-fast: false
```
Comment (mbostock) [inline .github/workflows/test.yml:15]: What does this do?

File: src/files.ts [hunk 2]
```diff
@@ -1,17 +1,27 @@
...
+}
+
+export function fromOsPath(path: string): string {
+  return path.split(op.sep).join(sep);
+}
+
 /**
  * Returns the relative path from the current working directory to the given
- * Framework source file, such as "./src/client/search.js". This is typically
+ * Framework source file, such as "./src/client/search.js". This is typigit pucally
```
Comment (mbostock) [inline src/files.ts:20]:
```suggestion
 * Framework source file, such as "./src/client/search.js". This is typically
```

File: .github/workflows/test.yml (hunk 1 above)
Comment (mythmon) [reply .github/workflows/test.yml:15]: `fail-fast: true` (the default) means that if any CI job fails, the rest are immediately cancelled. `fail-fast: false` lets them all run to completion. `true` saves on CI time, `false` leads to easier debugging.

**Comment by mythmon** [comment]
We need to add rimraf to the template package.json's

**Pull request merged**